scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,279 @@
+              parent[parentProperty].push(dataToMerge);
+            } else {
+              // Otherwise, deep merge the existing data with the new data
+              parent[parentProperty] = deepMerge(parent[parentProperty], dataToMerge);
+            }
+          });
+        }
+      }
+    } catch (e) {
+      console.error(`\nError processing update for '${update.name}' at path '${update.json_path}'`, e);
+    }
+  });
+  fs.writeFileSync(outputPath, JSON.stringify(baseDoc, null, 2), 'utf8');
+}
+
+// Iterate over the combinedData array and apply updates for each item
+const outputDir = path.join(__dirname, 'updated-docs');
+const validationPromises = []; // Array to store all validation promises
+combinedData.forEach((item) => {
+  const baseDocPath = item.name && item.name.includes("Security Scheme Object")
+    ? 'base-doc-security-scheme-object.json'
+    : 'base-doc.json';
+
+  // Check if the directory exists, and if not, create it
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(outputDir, { recursive: true });
+  }
+
+  const processedExampleName = item.name.replace(/[^\w\s-]/g, '').replace(/\s+/g, '-');
+  const outputPath = path.join(outputDir, `${processedExampleName}.json`);
+  // const outputPath = `./updated-docs/updated-doc-${index + 1}.json`;
+  // console.log(`\n${combinedData[num-1].name} = ${currentExample}`);
+
+  // Apply updates and save the document
+  applyUpdatesAndSave([item], baseDocPath, outputPath);
+
+  // Validate the output file using the AsyncAPI parser
+  // const validationPromise = validateParser(outputPath);
+
+  // Use the AsyncAPI CLI to validate the output file
+  const validationPromise = validateCli(outputPath)
+  .then((output) => {
+    console.log(`\n${output}`);
+  })
+  .catch((error) => {
+    console.error(error);
+  });
+  // Delete the updated document after saving
+  // fs.unlinkSync(outputPath);
+
+  validationPromises.push(validationPromise);
+});
+
+//  Function to validate a file using AsyncAPI parser
+async function validateParser(filePath) {
+  const document = fs.readFileSync(filePath, 'utf8');
+  try {
+    const diagnostics = await parser.validate(document);
+
+    if (diagnostics.length > 0) {
+      diagnostics.forEach(diagnostic => {
+        if (diagnostic.level === 'error') {
+          console.error(`Error in ${filePath}: ${diagnostic.message}`);
+        } else {
+          console.log(`Warning in ${filePath}: ${diagnostic.message}`);
+        }
+      });
+    } else {
+      console.log(`${filePath} is valid.`);
+    }
+  } catch (error) {
+    console.error(`Validation failed for ${filePath}: ${error.message}`);
+  }
+}
+
+// Function to validate a file using AsyncAPI CLI
+async function validateCli(filePath) {
+  return new Promise((resolve, reject) => {
+    // Construct the command to run the AsyncAPI CLI validate command
+    const command = `npx asyncapi validate ${filePath}`;
```

Comment:
You should definitely use the parser directly and not call a cmd command here.

Reply:
Got it.

Reply:
resolve via d7eb944